LanguageFeatures/Wildcards/binding_A01_t11.dart

```diff
@@ -0,0 +1,403 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion A local declaration whose name is `_` does not bind that name to
+/// anything.
+///
+/// @description Checks that no entry named `_` is introduced into the enclosing
+/// scope by a wildcarded declaration. Test function literals calling
+/// constructors and functions tear-offs
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=wildcard-variables
+
+class A {
+  A(_);
+  A.constr([int _ = 1]);
+}
+
+void topLevelFunction1(int _) {}
+void topLevelFunction2([int _ = 1]) {}
+
+class C {
+  static void staticMethod1(int _) {}
+  static void staticMethod2([int _ = 2]) {}
+  void instanceMethod1(int _) {}
+  void instanceMethod2([int _ = 3]) {}
+}
+
+mixin M {
+  static void staticMethod1(int _) {}
+  static void staticMethod2([int _ = 4]) {}
+  void instanceMethod1(int _) {}
+  void instanceMethod2([int _ = 5]) {}
+}
+
+enum E {
+  e1;
+  static void staticMethod1(int _) {}
+  static void staticMethod2([int _ = 6]) {}
+  void instanceMethod1(int _) {}
+  void instanceMethod2([int _ = 7]) {}
+}
+
+extension type ET(int id) {
+  static void staticMethod1(int _) {}
+  static void staticMethod2([int _ = 8]) {}
+  void instanceMethod1(int _) {}
+  void instanceMethod2([int _ = 9]) {}
+}
+
+class D {}
+extension Ext on D {
+  static void staticMethod1(int _) {}
+  static void staticMethod2([int _ = 2]) {}
+  void instanceMethod1(int _) {}
+  void instanceMethod2([int _ = 3]) {}
+}
+
+class MA = Object with M;
+
+main() {
+  void localFunction1(int _) {}
+  void localFunction2([int _ = 12]) {}
+  var functionExpression1 = (int _) {};
+  var functionExpression2 = ([int _ = 13]) {};
+
+  var to1 = A.new;
+  var f1 = (_) {
```

Comment:
Hmmm, thinking... ;-)

I don't think all these function literals will test anything new, we already have other tests about function literals.

All we need to check for the tear-offs is that we can perform the tear-off and the tear-off can be called (and it is accepted at compile time as well as running without throwing at run time).

When I mentioned a function literal it was a reference to a possible implementation technique. The expression `A.new` evaluates to a function object at run time. A constructor is not a function object, and this means that the operation that tears off a constructor is an operation that creates a function object (from something that we don't need to know anything about, because the representation of a constructor as such is an implementation specific property).

However, with `A.new`, that function object would be very similar to the function objects that we can obtain by evaluating the function literal `(dynamic d) => A(d)`, and `A.constr` would be similar to `([int i = 1]) => A.constr(i)`.

However, I've renamed the variables here in order to make sure that the result has the right behavior. An implementation could (if someone forgot to think about it ;-) model those constructor tear-offs using the same variable names as in the declaration of the constructor:

If so, then `A.new` would be modeled by `(_) => A(_)`, which is an error. This could be an internal error at compile time, or it could be a run-time error.

All in all, we should have what we need by means of something like the following code:

```dart
void main() {
  ...
  var to1 = A.new;
  to1('Hello')..checkThatTheArgumentWasReceivedCorrectly;
  var to2 = A.constr;
  to2(15)..checkThatTheArgumentWasReceivedCorrectly;
  ...
}
```

I think this test could me adjusted to do this such that it has no compile-time errors, and such that we can see that the tear-off can be performed (no compile-time errors, no run-time errors), and the behavior of the constructor is as expected.

Most of it would amount to deleting a bunch of lines from this `main`, but it would be very good if it is also tested that the transfer of actual arguments from the function object to the constructor works as expected.